bql/planner/filter/filter.go

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package filter isolates core FILTER related implementation.
+package filter
+
+import (
+	"fmt"
+)
+
+// Operation represents a filter operation supported in BadWolf.
+type Operation int
+
+// List of supported filter operations.
+const (
+	Latest Operation = iota + 1
```

Comment:
Since `0` is the default `nil` value for an `int`, I did not want it to be confused with the first `filter.Operation` of our `Operations` list above (the `Latest` operation above).

This way, I can verify if `Operation` was indeed already set in a `FilterClause` (or if it still is `nil`), as I do in `WhereFilterClauseHook` for example:

```
if currFilter == nil || currFilter.Operation == filter.Operation(0) || currFilter.Binding == "" {
    return nil, fmt.Errorf("could not add invalid working filter %q to the statement filters list", currFilter)
}
st.AddWorkingFilterClause()
```

To make this clearer in code I can also add a `func (op Operation) isEmpty() bool` method (or something like this) in `filter.go`, that could abstract this comparison for us.

Reply:
I have just added this `isEmpty` method for `filter.Operation` too.